src/tyr/matrix_serializer.cc

```diff
@@ -121,18 +127,44 @@ json::ArrayPtr serialize_row(const std::vector<TimeDistance>& tds,
 json::MapPtr serialize(const Api& request,
                        const std::vector<TimeDistance>& time_distances,
                        double distance_scale) {
-  json::ArrayPtr matrix = json::array({});
+  auto json = json::map({});
   const auto& options = request.options();
-  for (size_t source_index = 0; source_index < options.sources_size(); ++source_index) {
-    matrix->emplace_back(serialize_row(time_distances, source_index * options.targets_size(),
-                                       options.targets_size(), source_index, 0, distance_scale));
+
+  if (options.verbose()) {
```

Comment:
Almost, but not quite. If you read https://github.com/valhalla/valhalla/issues/3866#issuecomment-1346362590 and https://github.com/valhalla/valhalla/issues/3866#issuecomment-1442410029 carefully, it says what to do about the `verbose` option: we want to default it right here to `true` to keep backwards-compatibility (no one/client should be screwed once this PR is merged). However, if smth else is requested, we want to keep the current behavior of `verbose` to default to false.

Reply:
I got it, unless the user request with `"verbose": false` in the request JSON we shouldn't trim the response JSON.

But, here `verbose` defaults to `false`. How can we make it default `true'? 
 https://github.com/valhalla/valhalla/blob/master/proto/options.proto#L396 
Doesn't it will affect `/locate` and `/status` requests?

Reply:
Sorry, I wasn't expressing myself very clearly: in this line we need to check whether `verbose` was explicitly set to false. Make the protobuf `verbose` a `oneof` (which is safe for `/locate` and others) and check here if `options.has_verbose_case() && !options.verbose()` (and obviously turn the body of if/else around). Or something like that, not tested.

Reply:
> Make the protobuf `verbose` a `oneof`
didn't get this line :)

Can we have a call on Discord or something to clear the whole thing?

Reply:
Oops, it's already a `oneof` :sweat_smile: Then ignore that particular comment.

Reply:
Need help!
`options.has_verbose()` and `options.has_verbose_case()` are always true, whether we pass in request json `"verbose": false` or doesn't pass `"verbose"`

Spend a lot of time :)

Reply:
yeah, it needs to be looked at when the json is being parsed. had to look myself, see [`aec7f58` (#3987)](https://github.com/valhalla/valhalla/pull/3987/commits/aec7f58bae6eb7e0a76c85a4bd72c6f698c6307d)

Reply:
Yay! Working, finally!

vee_nits123@Veenits-Air conf % valhalla_service valhalla.json sources_to_targets '{"sources":[{"lat":41.318818,"lon":19.461336}],"targets":[{"lat":41.318818,"lon":19.461336},{"lat":41.321001,"lon":19.459598}],"costing":"auto"}'
```json
{"units":"kilometers","sources":[[{"lon":19.461336,"lat":41.318818}]],"targets":[[{"lon":19.461336,"lat":41.318818},{"lon":19.459598,"lat":41.321001}]],"sources_to_targets":[[{"distance":0.000,"time":0,"to_index":0,"from_index":0},{"distance":0.287,"time":21,"to_index":1,"from_index":0}]]}
```
vee_nits123@Veenits-Air conf % valhalla_service valhalla.json sources_to_targets '{"sources":[{"lat":41.318818,"lon":19.461336}],"targets":[{"lat":41.318818,"lon":19.461336},{"lat":41.321001,"lon":19.459598}],"costing":"auto","verbose":false}'
```json
{"units":"kilometers","sources_to_targets":{"durations":[[0,21]],"distances":[[0.000,0.287]]}}
```
vee_nits123@Veenits-Air conf % valhalla_service valhalla.json sources_to_targets '{"sources":[{"lat":41.318818,"lon":19.461336}],"targets":[{"lat":41.318818,"lon":19.461336},{"lat":41.321001,"lon":19.459598}],"costing":"auto","verbose":true}' 
```json
{"units":"kilometers","sources":[[{"lon":19.461336,"lat":41.318818}]],"targets":[[{"lon":19.461336,"lat":41.318818},{"lon":19.459598,"lat":41.321001}]],"sources_to_targets":[[{"distance":0.000,"time":0,"to_index":0,"from_index":0},{"distance":0.287,"time":21,"to_index":1,"from_index":0}]]}
```